lib/ansible/modules/cloud/openstack/os_server_tag.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# (c) 2018, Artem Goncharov <artem.goncharov@gmail.com>
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: os_server_tag
+short_description: Manage server tags
+extends_documentation_fragment: openstack
+version_added: "2.8"
+author: "Artem Goncharov (gtema)"
+description:
+    - set or delete tag on the Nova server
+options:
+   server:
+      description:
+        - Name or id of the server
+      required: true
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+   tag:
+     description:
+       - tag name
+     required: true
+   availability_zone:
+     description:
+       - Ignored. Present for backwards compatibility
+requirements:
+    - "python >= 2.7"
+    - "openstacksdk"
+'''
+EXAMPLES = '''
+---
+- name: set the tag on the instance
+  ignore_errors: True
+  os_server_tag:
+    server: "{{ server_name }}"
+    state: present
+    name: __crazy_tag
+'''
+
+RETURN = '''
+server:
+    description: UUID of the server.
+    returned: success
+    type: string
+changed:
+    description: Identify whether request has been sent.
+    returned: always
+    type: boolean
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.openstack import openstack_full_argument_spec, openstack_module_kwargs, openstack_cloud_from_module
+from ansible.module_utils._text import to_native
+
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        server=dict(required=True),
+        state=dict(default='present', choices=['absent', 'present']),
+        tag=dict(required=True),
+    )
+
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec,
+                           supports_check_mode=True,
```

Comment:
I don't see any check mode support implemented in the code, is there?

Reply:
Nope, removed